src/main.rs

```diff
@@ -1,98 +1,173 @@
-use std::convert::TryInto;
+use std::env;
 
-const BLOCK_SIZE: usize = 32; // Size of each block in bits
-const HASH_SIZE: usize = 32; // Size of the hash code in bits
+const BLOCK_SIZE: usize = 8; // Size of each block in bytes
+const HASH_SIZE: usize = 8; // Size of the hash code in bytes
+const INITIAL_STATE: [u8; HASH_SIZE] = [255, 100, 211, 37, 112, 167, 41, 37]; // randomly generated initial state
```

Comment:
I just used a random number generator. SHA2 uses something clever like the square root of prime numbers. Maybe copy what they did.